app/src/main/java/de/grobox/transportr/settings/SettingsFragment.kt

```diff
@@ -91,6 +103,37 @@ class SettingsFragment : PreferenceFragmentCompat() {
                 true
             }
         }
+
+        arrayOf(PROXY_ENABLE, PROXY_HOST, PROXY_PORT, PROXY_PROTOCOL).forEach { prefKey ->
+            (findPreference(prefKey) as Preference?)?.let { pref ->
+                pref.setOnPreferenceChangeListener { _, newValue ->
+                    try {
+                        val newProxy = settingsManager.getProxy(mapOf(
+                            Pair(prefKey, newValue)
+                        ))
+                        TransportrUtils.checkInternetConnectionViaProxy(newProxy)
+                        TransportrUtils.updateGlobalHttpProxy(newProxy, manager)
+                    } catch (e: Exception) {
+                        Log.e(TAG, "Invalid proxy settings: " + e.message)
+                        AlertDialog.Builder(context!!)
+                            .setTitle(R.string.invalid_proxy_settings)
+                            .setMessage(e.message)
+                            .setPositiveButton(android.R.string.ok) { di, _ -> di.dismiss() }
+                            .create()
+                            .show()
+                    }
+                    true
+                }
+            }
+        }
+
+        findPreference<ValidatedEditTextPreference>(PROXY_HOST)!!.validate = { value ->
+            Patterns.DOMAIN_NAME.matcher(value).matches() || Patterns.IP_ADDRESS.matcher(value).matches()
```

Comment:
Perhaps we should add "localhost" then manually as I would see it as a common setting?